internal/storage/v1/elasticsearch/spanstore/writer.go

```diff
@@ -118,6 +118,7 @@ func getSpanAndServiceIndexFn(p SpanWriterParams, writeAlias string) spanAndServ
 func (s *SpanWriter) WriteSpan(spanStartTime time.Time, span *dbmodel.Span) {
 	s.writerMetrics.Attempts.Inc(1)
 	s.convertNestedTagsToFieldTags(span)
+	s.promoteSpanKindFromTags(span)
```

Comment:
I think this should happen in writer_v1 when creating dbmodel object. In this lower-level writer we should be able to assume that the span.kind has been already extracted to a proper field in dbmodel.Span